packages/@aws-cdk/aws-lambda/test/test.lambda.ts

```diff
@@ -1219,66 +1219,101 @@ export = {
+              Resource: { 'Fn::GetAtt': ['Function76856677', 'Arn'] },
+            },
+          ),
+        }),
+        Roles: ['someRole'],
+      }));
 
-    test.done();
-  },
+      test.done();
+    },
 
-  'grantInvoke with an imported role (from a different account)'(test: Test) {
-    // GIVEN
-    const stack = new cdk.Stack(undefined, undefined, {
-      env: { account: '3333' },
-    });
-    const fn = new lambda.Function(stack, 'Function', {
-      code: lambda.Code.fromInline('xxx'),
-      handler: 'index.handler',
-      runtime: lambda.Runtime.NODEJS_10_X,
-    });
+    'with an imported role (from a different account)'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack(undefined, undefined, {
+        env: { account: '3333' },
+      });
+      const fn = new lambda.Function(stack, 'Function', {
+        code: lambda.Code.fromInline('xxx'),
+        handler: 'index.handler',
+        runtime: lambda.Runtime.NODEJS_10_X,
+      });
 
-    // WHEN
-    fn.grantInvoke(iam.Role.fromRoleArn(stack, 'ForeignRole', 'arn:aws:iam::123456789012:role/someRole'));
+      // WHEN
+      fn.grantInvoke(iam.Role.fromRoleArn(stack, 'ForeignRole', 'arn:aws:iam::123456789012:role/someRole'));
 
-    // THEN
-    expect(stack).to(haveResource('AWS::Lambda::Permission', {
-      Action: 'lambda:InvokeFunction',
-      FunctionName: {
-        'Fn::GetAtt': [
-          'Function76856677',
-          'Arn',
-        ],
-      },
-      Principal: 'arn:aws:iam::123456789012:role/someRole',
-    }));
+      // THEN
+      expect(stack).to(haveResource('AWS::Lambda::Permission', {
+        Action: 'lambda:InvokeFunction',
+        FunctionName: {
+          'Fn::GetAtt': [
+            'Function76856677',
+            'Arn',
+          ],
+        },
+        Principal: 'arn:aws:iam::123456789012:role/someRole',
+      }));
+
+      test.done();
+    },
+
+    'on an imported function (same account)'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack(undefined, undefined, {
+        env: { account: '123456789012' },
+      });
+      const fn = lambda.Function.fromFunctionArn(stack, 'Function', 'arn:aws:lambda:us-east-1:123456789012:function:MyFn');
+
+      // WHEN
+      fn.grantInvoke(new iam.ServicePrincipal('elasticloadbalancing.amazonaws.com'));
+
+      // THEN
+      expect(stack).to(haveResource('AWS::Lambda::Permission', {
+        Action: 'lambda:InvokeFunction',
+        FunctionName: 'arn:aws:lambda:us-east-1:123456789012:function:MyFn',
+        Principal: 'elasticloadbalancing.amazonaws.com',
+      }));
+
+      test.done();
+    },
+
+    'on an imported function (different account)'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack(); // Unresolved account
+      const fn = lambda.Function.fromFunctionArn(stack, 'Function', 'arn:aws:lambda:us-east-1:123456789012:function:MyFn');
+
+      // WHEN
+      fn.grantInvoke(new iam.ServicePrincipal('elasticloadbalancing.amazonaws.com'));
```

Comment:
Are we sure this shouldn't throw (as something that doesn't make sense?)

Can we add least add a `warning` to the construct tree or something?

Reply:
It's an unusual case, certainly. Effectively used by power users who already know their function has appropriate permissions and just need to add the reference to the target group(s). It certainly shouldn't fail with the current `findChild` error. Adding a warning to the tree makes sense.